### DISCUSSION TITLE
How to start creating maskbook plugins?

### DISCUSSION THREAD

**Discussion by kecrily** [DimensionDev/Maskbook#3711] [Q&A] [answered]
I'm tring to create a maskbook plugin. But I ran into some problems.

After executing `pnpm install` and `pnpm start`, I copied the folder in `packages/plugins/example/src` into `packages/maskbook/src/plugins` and change the information in `base.ts`.

Following the documentation guidelines in the `docs` directory, I loaded the generated `dist` directory in Chrome. However, I opened the Dashboard and did not find the plugin that corresponds to my modified information.

Some other information that may be useful:

- Node: 16.4.2
- pnpm: 6.10.2
- Chrome: 91.0.4472.164 (x64)
- Windows: 11 22000.71 (x64)
- clone version: [`0c90cfe`](https://github.com/DimensionDev/Maskbook/commit/0c90cfe45f065e0ea8c5ba5950da41e7b38c6c6a)

**Comment by septs** [comment]
https://github.com/DimensionDev/Maskbook/blob/develop/docs/plugin-development-guide.md

**Reply by kecrily** [reply]
You may not have read what I posted carefully, I did mention that I had read the documentation.

**Comment by Jack-Works** [comment]
Did you registered your new plugin? It won't be automatically loaded. A "import '....'" to your index.ts is necessary.

**Reply by kecrily** [reply]
Thank you for the reminder. But that's not the reason for my problem, the `registerPluginis` included in the code of example.

I found that I also need to register in plugin-infra/registy.ts

**Comment by kecrily** [comment, marked as answer]
If you have the same problem as me, go to `packages\maskbook\src\plugin-infra\register.ts` and import your plugin.

And

```
// If your plugin also works in isolated dashboard, please also register it in
// packages/dashboard/src/initialization/plugins.ts
```

**Comment by Jack-Works** [comment]
Improved docs at https://github.com/DimensionDev/Maskbook/pull/3715. Thanks.
